eng/scripts/build.ps1

```diff
@@ -35,10 +37,19 @@ $keys | ForEach-Object { $sdks[$_] } | ForEach-Object {
     if ($_.generate) {
         Write-Host "##[command]Executing autorest.go in " $_.path
         $autorestPath = $_.path + "\autorest.md"
+
+        if (ShouldGenerate-AutorestConfig($autorestPath) ) {
```

Comment:
powershell functions should not be called with parentheses (e.g. this should be `if (ShouldGenerate-AutorestConfig $autorestPath) {}`.

What's actually happening here is powershell is treating the parentheses as a higher precedence expression and just resolving the variable. This works in this case because the function only takes a single argument. It gets more tricky with multiple arguments. For example, the below would parse the parenthesized expression as a powershell array:

```
$ function foo($a, $b) { echo "a: " + $a; echo "b: " + $b }
$ foo("a", "b")
a: a b
b:
$ foo "a" "b"
a: a
b: b
```